shaders/clouds.omwfx

```diff
@@ -513,9 +502,25 @@ fragment main {
             * max(sun_dir.z, 0.0) * 3.0
             // Regulate direct lighting with max_dist to prevent it appearing on top of close objects
             * min(max_dist * (1.0 / 100000.0), 1.0);
-        vec3 fog_color = (sun_light + moon_light) * mix(sky_color, sky_tone, max(sun_dir.z, 0.3));
-
-        vec3 fog_light = fog_color + direct_light * sun_color;
+        //Define the atmospheric attenuation factors for sunlight's red, green, and blue channels
+        vec3 att_factor = vec3(
+            (pow((2 / (abs(sun_dir.z) + 1)), 2) / 2),
+            (sqrt(1 / (abs(sun_dir.z))) / 10 * (abs(sun_dir.z))),
+            (pow((1 / (abs(sun_dir.z) + 1)), 2) * 3 * (-1 * (abs(sun_dir.z))))
+        );
+        //Interpolate between original sun color and attenuated color based on sun's proximity to the horizon
+        vec3 sun_attenuated = vec3(mix(vec3(min((sun_color.x * ((dot(dir, sun_dir) + 1) / 2) * att_factor.x), 255), min((sun_color.y * (dot(dir, sun_dir) + 1) * att_factor.y), 255), min((sun_color.z / (dot(dir, sun_dir) + 2) * att_factor.z),255)), sun_color, abs(sun_dir.z)));
+        //Have fog color be directionally influenced by sunlight
+        vec3 fog_color = mix(
+            vec3(omw.fogColor), 
+            (mix
+                (mix(sky_color, sun_attenuated, max(dot(dir,sun_dir), 0.0001)), 
+                (sun_attenuated + sky_color), 
+                pow(max(dot(dir,sun_dir), 0.0001),3))
+            ), 
+            (sqrt(max(dot(dir,sun_dir), 0.0001))) / 2);
```

Comment:
Prefer `* 0.5` if possible: multiplication is far faster than division and since floating point ops are subject to precision issues, many compilers won't (and shouldn't) perform this optimisation themselves.